src/wp-includes/comment-template.php

```diff
@@ -233,7 +233,13 @@ function get_comment_author_email_link( $link_text = '', $before = '', $after =
 function get_comment_author_link( $comment_id = 0 ) {
 	$comment = get_comment( $comment_id );
 
-	$comment_id = ! empty( $comment->comment_ID ) ? $comment->comment_ID : (string) $comment_id;
+	if ( ! empty( $comment->comment_ID ) ) {
+		$comment_id = $comment->comment_ID;
+	} elseif ( is_scalar( $comment_id ) ) {
+		$comment_id = (string) $comment_id;
+	} else{
```

Comment:
```suggestion
	} else {
```

Reply:
@mukeshpanchal27 Thanks for the review, suggested change has been made.